### PR TITLE
Support for saving and restoring workspaces by number

### PR DESCRIPTION
**Pull request by cedl38**
This pull request allow selecting workspace by number.

When a number is provided to --workspace option, the workspace that match the corresponding number is selected. It work's for save and restore commands.

### CODE REVIEW

File: i3_resurrect/treeutils.py [hunk 1]
```diff
@@ -88,8 +88,14 @@ def get_workspace_tree(workspace):
             if container['type'] != 'con':
                 pass
             for ws in container['nodes']:
-                if ws['name'] == workspace:
-                    return ws
+                # Select workspace and trigger name and num field
+                if workspace.isdigit():
+                    if 'num' in ws:
+                        if ws['num'] == int(workspace):
+                            return ws
```
Comment (JonnyHaystack) [inline i3_resurrect/treeutils.py]: Would rather avoid these extra indentation levels and use `and` instead

File: i3_resurrect/treeutils.py [hunk 2]
```diff
@@ -88,8 +88,14 @@ def get_workspace_tree(workspace):
...
+                else:
+                    if ws['name'] == workspace:
```
Comment (JonnyHaystack) [inline i3_resurrect/treeutils.py]: Similarly here I would opt for `elif` instead of an extra unnecessary indentation level

Comment (cedl38) [reply i3_resurrect/treeutils.py]: Ok, it's clearer indeed. 
I updated pull request.

Also, Your plugin is great, thanks very much for your work!

**Comment by JonnyHaystack** [comment]
Other than the minor comments I've left, I have another concern. This change makes it so if the workspace name is numeric, it will match the first workspace is in the tree with that number. This prompts me to ask the following question:
What if the first "num" match is not the closest "name" match? Consider the following scenario:
- You go to workspace "10". Its name and number are both 10.
- Open a window there to keep it open, then go to workspace "10:". Its name is "10:" but its number is 10.
- When trying this locally I found that workspace "10:" ended up being before workspace "10" in the tree, and so `i3-msg workspace number 10` takes me to workspace "10:", not workspace "10".
- Now at this point with your patch, `i3-resurrect save -w 10` would save workspace "10:" not workspace "10", and you would have no way to save workspace "10".

So it may be necessary to have an option to specify numeric mode. Maybe something like `i3-resurrect save -n -w 10`. That way it will behave like `i3-msg workspace number N` only when people want it to, and otherwise would still work as expected.

**Comment by cedl38** [comment]
@JonnyHaystack  Yes I'm aware about this behavior.
I have considered to add extra option but thinking to avoid unnecessary extra complexity.
Your proposal to add -n option like `save -n -w 10` seem to be good to avoid confusion

**Comment by JonnyHaystack** [comment]
Okay I can put that option in when I do the merge then. Thanks!

**Pull request merged**